File: samples/python/change_lane.py
# Import module
import pyscaner

# Create API object
api = pyscaner.API("EASYSCANER", "DEFAULT_1.9", 10)
tick = 0

# Define the initialisation function
def init():
    for v in api.get_vehicles():
        v.try_pull_out()
    print("All vehicles were asked to pull out")

# Set the initialisation function
api.on_init(init)

# Define the loop function
def running():
    global tick
    tick += 1
    print(tick)

    if tick == 5:
        for v in api.get_vehicles():
            v.try_filter_in()
        print("All vehicles were asked to filter in")
        api.while_running(None)

# Set the loop function
api.while_running(running)

api.run()

File: samples/EasyScaner/ChangeLane.cpp
// Include EasyScaner files
#include "EasyScaner/Scaner.hpp"
#include "EasyScaner/Vehicle.hpp"

// Main function; setup goes here
int main(int argc, char* argv[]) {
  // Create Scaner object
  Scaner sc = Scaner("EASYSCANER", "CONFIG", 1);
  int tick;

  // Set the initialisation function
  sc.on_init([&]() {
    tick = 0;
    for (auto vhl : sc.get_road_vehicles()) vhl->try_pull_out();  // Ask each vehicle to move a lane to the left
    sc.log("All vehicles have been ordered to move to the lane on their left.");
  });

  // Set the loop function
  sc.while_running([&]() {
    if (tick++ == 5) {                                               // If 5 SCANeR loops have passed
      for (auto vhl : sc.get_road_vehicles()) vhl->try_filter_in();  // Ask each vehicle to move a lane to the right
      sc.log("All vehicles have been ordered to move to the lane on their right.");
      sc.while_running([&]() {});  // Stop the loop from running by setting an empty function as callback
    }
  });

  // Run the tool
  sc.run();
}
